pkg/harvester/edit/loadbalancer.harvesterhci.io.ippool/PriorityRow.vue

```diff
@@ -0,0 +1,227 @@
+
+    namespaceOptions() {
+      const namespaces = this.$store.getters['harvester/all'](NAMESPACE);
+
+      const out = namespaces.filter((namespace) => {
+        if (this.row.project === '*') {
+          return true;
+        } else if (this.row.project) {
+          return namespace.project?.id === this.row.project;
+        } else {
+          return true;
+        }
+      }).map((namespace) => {
+        return {
+          label: namespace.metadata.name,
+          value: namespace.id,
+        };
+      });
+
+      return [{
+        label: this.t('generic.all'),
+        value: '*',
+      }, ...out];
+    },
+
+    guestClusterOptions() {
+      const clusters = this.$store.getters['management/all'](CAPI.RANCHER_CLUSTER);
+      const configs = this.$store.getters['management/all'](HCI.HARVESTER_CONFIG);
+      const selectedClusters = this.rows.map(row => row?.guestCluster);
+
+      const out = clusters.filter((c) => {
+        if (this.row.namespace === '*') {
+          return true;
+        }
+
+        const machinePools = c.spec?.rkeConfig?.machinePools || [];
+        const machineConfigName = machinePools[0]?.machineConfigRef?.name;
+        const config = configs.find(c => c.id === `fleet-default/${ machineConfigName }`);
+
+        if (config) {
+          const vmNamespace = config?.vmNamespace;
+
+          return vmNamespace === this.row.namespace && !selectedClusters.includes(c.id);
+        } else {
+          return false;
+        }
+      }).map((cluster) => {
+        return {
+          label: cluster.nameDisplay,
+          value: cluster.metadata.name,
+        };
+      });
+
+      return [{
+        label: this.t('generic.all'),
+        value: '*',
+      }, ...out];
+    },
+
+    projectOptions() {
+      const projects = this.$store.getters['management/all'](MANAGEMENT.PROJECT);
+
+      const out = projects.filter(p => p.metadata.namespace === this.currentCluster.id).map((project) => {
+        return {
+          label: project.nameDisplay,
+          value: project.id,
+        };
+      });
+
+      return [{
+        label: this.t('generic.none'),
+        value: '',
+      }, {
+        label: this.t('generic.all'),
+        value: '*',
+      }, ...out];
+    },
+  },
+
+  methods: {
+    update() {
+      const { namespace, project, guestCluster } = this.row;
+
+      this.$emit('input', {
+        namespace,
+        project,
+        guestCluster,
+      });
+    },
+
+    remove() {
+      this.$emit('remove');
+    },
+  },
+
+  watch: {
+    'row.project'() {
+      if (this.row.namespace !== '*') {
+        this.row.namespace = '';
+      }
+
+      if (this.row.guestCluster !== '*') {
+        this.row.guestCluster = '';
+      }
+    },
+
+    'row.namespace'() {
+      if (this.row.guestCluster !== '*') {
+        this.row.guestCluster = '';
+      }
+    },
+  },
+};
+</script>
+
+<template>
+  <div
+    class="pool-row"
+    :class="{
+      'show-project': !isSingleVirtualCluster,
```

Comment:
<img width="1593" alt="image" src="https://github.com/harvester/dashboard/assets/24985926/8bc0389b-9d95-4246-8c8d-74a895650324">

Perhaps "Project" should not be displayed here?

Reply:
Done